spec/models/publication_spec.rb

```diff
@@ -252,6 +252,46 @@
     end
   end
 
+  describe 'update_from_pubmed' do
+    it 'should not update from pubmed source if there is no pmid' do
+      expect(publication.pmid).to be_nil
+      expect(publication.update_from_pubmed).to be false
+    end
+    it 'should update from pubmed source if there is a pmid' do
+      pmid = 1
+      publication.pmid = pmid
+      source_data = '<PubmedArticle><MedlineCitation Status="Publisher" Owner="NLM"><PMID Version="1">1</PMID><OriginalData/></PubmedArticle>'
+      new_source_data = '<PubmedArticle><MedlineCitation Status="Publisher" Owner="NLM"><PMID Version="1">1</PMID><SomeNewData/></PubmedArticle>'
+      pubmed_record = PubmedSourceRecord.create(pmid: pmid, source_data: source_data)
+      allow(PubmedSourceRecord).to receive(:find_by_pmid).with(pmid).and_return(pubmed_record)
+      expect(pubmed_record.source_data).to be_equivalent_to source_data
+      allow_any_instance_of(PubmedClient).to receive(:fetch_records_for_pmid_list).with(pmid).and_return(new_source_data)
+      expect(publication.pmid).to_not be_nil
+      expect(publication.update_from_pubmed).to be true
+      expect(PubmedSourceRecord.find_by_pmid(pmid).source_data).to be_equivalent_to new_source_data
+    end
+  end
+
+  describe 'update_from_sciencewire' do
+    it 'should not update from sciencewire source if there is no sciencewire_id' do
+      expect(publication.sciencewire_id).to be_nil
+      expect(publication.update_from_sciencewire).to be false
+    end
+    it 'should update from sciencewire source if there is a sciencewire_id' do
+      sciencewire_id = 1
+      publication.sciencewire_id = sciencewire_id
+      source_data = '<?xml version="1.0"?><PublicationItem><PublicationItemID>1</PublicationItemID><Title>Some Title</Title><Abstract/></PublicationItem>'
+      new_source_data = '<?xml version="1.0"?><PublicationItem><PublicationItemID>1</PublicationItemID><Title>Some New Title</Title><Abstract/></PublicationItem>'
+      sw_record = SciencewireSourceRecord.create(sciencewire_id: sciencewire_id, source_data: source_data)
+      allow(SciencewireSourceRecord).to receive(:find_by_sciencewire_id).with(sciencewire_id).and_return(sw_record)
+      expect(sw_record.source_data).to be_equivalent_to source_data
+      allow_any_instance_of(ScienceWireClient).to receive(:get_sw_xml_source_for_sw_id).with(sciencewire_id).and_return(Nokogiri::XML(new_source_data).xpath('//PublicationItem').first)
+      expect(publication.sciencewire_id).to_not be_nil
+      expect(publication.update_from_sciencewire).to be true
+      expect(SciencewireSourceRecord.find_by_sciencewire_id(sciencewire_id).source_data).to be_equivalent_to new_source_data
```

Comment:
but that wouldn't happen until the rebuild_pub_hash step, which i would argue is not what this tests is looking for

I checked and sadly "rebuild_pub_hash" is one of the pending tests... and while I think it would be great to add tests for that, I don't think I have the time to do that before needing to use this code.  I did a ticket for it though : https://github.com/sul-dlss/sul_pub/issues/180

ditto for pubmed source record test